io/FV3GFS_io.F90

```diff
@@ -783,7 +783,11 @@ subroutine sfc_prop_restart_read (Sfcprop, Atm_block, Model, fv_domain, warm_sta
       !--- allocate the various containers needed for restarts
       allocate(sfc_name2(nvar_s2m+nvar_s2o+nvar_s2mp+nvar_s2r))
       allocate(sfc_name3(0:nvar_s3+nvar_s3mp))
-      allocate(sfc_var2(nx,ny,nvar_s2m+nvar_s2o+nvar_s2mp+nvar_s2r),sfc_var3ice(nx,ny,Model%kice))
+      allocate(sfc_var2(nx,ny,nvar_s2m+nvar_s2o+nvar_s2mp+nvar_s2r))
+      ! Note that this may cause problems with RUC LSM for coldstart runs from GFS data
+      ! if the initial conditions do contain this variable, because Model%kice is 9 for
+      ! RUC LSM, but tiice in the initial conditions will only have two vertical layers
+      allocate(sfc_var3ice(nx,ny,Model%kice))
```

Comment:
So the current RUC LSM test does not use GFS data and its cold start ICs has 9 levels?

Reply:
No, this refers to the ice data. GFS v16 ICs do not contain the ice subsurface temperature
```
Sfcprop(nb)%tiice(ix,k) = sfc_var3ice(i,j,k)   !--- internal ice temp
```
in the coldstart data. Should, however, the tiice data from GFS sfc_sice physics be added to the coldstart ICs, then it would have only two layers, whereas RUC expects nine layers.

For restart/warmstart runs using the same physics, this is not a problem. tiice for RUC will be written with 9 layers and it will be read back in just fine.